src/Microsoft.AspNetCore.Mvc.Formatters.Json/JsonInputFormatter.cs

```diff
@@ -98,6 +128,22 @@ public class JsonInputFormatter : TextInputFormatter
             }
 
             var request = context.HttpContext.Request;
+
+            if (request.Body.CanSeek)
+            {
+                request.Body.Seek(0L, SeekOrigin.Begin);
```

Comment:
We should do this. 

Notice the follow is actually:
1. Enable buffering
2. Read to the end
3. Rewind to the beginning

We do 2, so we should do 3.